symengine/visitor.cpp

```diff
@@ -175,4 +177,195 @@ void TransformVisitor::bvisit(const MultiArgFunction &x)
     auto nbarg = x.create(newargs);
     result_ = nbarg;
 }
+
+class IsALinearArgTrigVisitor : public BaseVisitor<IsALinearArgTrigVisitor>
```

Comment:
Use `StopVisitor` and `preorder_traversal_stop` here.

Reply:
`StopVisitor` is not useful for stopping only locally. What I mean by this, is that If I want to break without seeing arguments of an `expr`, I can do only a global break, which might not be what is desired.
ex. - tan(x) + x
Add(Tan(x), x)
So, After checking tan(x), If I want to skip its arguments and go on to next Add argument I won't be able to do it using these two.

Reply:
Done.

Reply:
@isuruf , I need to skip calling `visit` on arguments of Trig functions like 
`sin(add(x, y))` , Once I see `sin` with arguments linear in `x`, I should skip the preorder_traversal in only that subtree and continue to the next subtree. How can I do this?

Reply:
I introduced a `preorder_traversal_local_stop` which can skip traversing the subtree generated by its arguments. Please let me know if that's Ok.